dbms/src/Interpreters/Join.cpp

```diff
@@ -900,9 +902,24 @@ Block Join::doJoinBlockHash(ProbeProcessInfo & probe_process_info) const
                 auto * current = reinterpret_cast<RowRefListWithUsedFlag *>(ptr_value);
                 current->setUsed();
             }
+        }
+        if (isRightSemiFamily(kind))
+        {
             // Return build table header for right semi/anti join
             block = sample_block_with_columns_to_add;
         }
+        else if (kind == ASTTableJoin::Kind::RightOuter)
+        {
+            block.erase(flag_mapped_entry_helper_name);
+            if (!non_equal_conditions.other_cond_name.empty())
+            {
+                block.erase(non_equal_conditions.other_cond_name);
+            }
+            if (!non_equal_conditions.other_eq_cond_from_in_name.empty())
+            {
+                block.erase(non_equal_conditions.other_eq_cond_from_in_name);
+            }
+        }
```

Comment:
Move to inside `if (useRowFlaggedHashMapIfHasOtherCondition(kind))`?

Reply:
ok